pulp_ansible/tests/functional/api/collection/v3/test_collection.py

```diff
@@ -62,15 +63,25 @@ def artifact():
     return artifact
 
 
+def get_metadata_published(pulp_client, pulp_dist):
+    """Return published datetime."""
+    metadata = pulp_client.using_handler(api.json_handler).get(
+        get_galaxy_url(pulp_dist["base_path"], "/v3/")
+    )
+    return datetime.strptime(metadata["published"], "%Y-%m-%dT%H:%M:%S.%fZ")
+
+
 @pytest.fixture(scope="session")
 def collection_upload(pulp_client, artifact, pulp_dist):
     """Publish a new collection and return the processed response data."""
     UPLOAD_PATH = get_galaxy_url(pulp_dist["base_path"], "/v3/artifacts/collections/")
-
+    published_before_upload = get_metadata_published(pulp_client, pulp_dist)
     logging.info(f"Uploading collection to '{UPLOAD_PATH}'...")
     collection = {"file": (ANSIBLE_COLLECTION_FILE_NAME, open(artifact.filename, "rb"))}
 
     response = pulp_client.using_handler(upload_handler).post(UPLOAD_PATH, files=collection)
+    published_after_upload = get_metadata_published(pulp_client, pulp_dist)
+    assert published_after_upload > published_before_upload
```

Comment:
@bmbouter is this enough or should I do a specific test?

Reply:
This is ok, but you need to use the test library to assert and not `assert` which could be disabled depending on how Python is compiled.

Reply:
so this entire file is affected, all the assertions are done like this

Reply:
yes none of them should use `assert`.

Reply:
apparently pytest rewrites the assertions: https://docs.pytest.org/en/2.1.0/assert.html#advanced-assertion-introspection

Reply:
Wow that's pretty amazing. Still though do we want to rely on this? I'm ok if so but I want to ask.

Reply:
@fao89 how about we merge this now since the other code uses this style already and would you be willing to put it on the next pulp_ansible weekly agenda?

Reply:
sounds good to me